### PR TITLE
Avoid suppressing git command output

### PR DESCRIPTION
**Pull request by rofinn**
Mostly replaces `success(cmd) || error("...")` in favour of just `run(cmd)`, so that we can produce useful errors for debugging git command failures.

### CODE REVIEW

**Comment by rofinn** [comment]
Nightly failures are unrelated.

File: src/Documenter.jl [hunk 1]
```diff
@@ -444,60 +444,57 @@ function deploydocs(;
...
-                        try run(`git commit -m "build based on $sha"`) end
+                            # Create the versions.js file containing a list of all docs
+                            # versions. This must always happen after the folder copying.
+                            Writers.HTMLWriter.generate_version_file(dirname)
 
-                        success(`git push -q upstream HEAD:$branch`) ||
-                            error("could not push to remote repo.")
+                            # Add, commit, and push the docs to the remote.
+                            run(`git add -A .`)
+                            try run(`git commit -m "build based on $sha"`) end
```
[the remark this answers is not in the document]
Comment (rofinn) [reply src/Documenter.jl]: Yeah, that seems much clearer. I wasn't sure what that was for and didn't want to break anything.

File: src/Documenter.jl [hunk 2]
```diff
@@ -444,60 +444,57 @@ function deploydocs(;
...
-                            Writers.HTMLWriter.generate_siteinfo_file(tagged_dir, travis_tag)
-                            # Build a `release-*.*` folder as well when the travis tag is
-                            # valid, which it *should* always be anyway.
-                            if ismatch(Base.VERSION_REGEX, travis_tag)
-                                version = VersionNumber(travis_tag)
-                                release = "release-$(version.major).$(version.minor)"
-                                gitrm_copy(target_dir, joinpath(dirname, release))
-                                Writers.HTMLWriter.generate_siteinfo_file(joinpath(dirname, release), release)
+                        try
+                            # Setup git.
```
[the remark this answers is not in the document]
Comment (rofinn) [reply src/Documenter.jl]: I was attempting to make minimal changes to the existing logic, but I'm happy to refactor this into separate function if you don't mind dealing with a larger PR.

[the remark this answers is not in the document]
Comment (mortenpi) [reply src/Documenter.jl]: A larger PR would not be a problem -- always happy to have things tidier.

File: src/Documenter.jl [hunk 3]
```diff
@@ -509,6 +433,102 @@ function deploydocs(;
...
+    latest_dir = joinpath(dirname, "latest")
+    stable_dir = joinpath(dirname, "stable")
+    tagged_dir = joinpath(dirname, tag)
+
+    keyfile = abspath(joinpath(root, ".documenter"))
+    target_dir = abspath(target)
+
+    upstream = "git@$(replace(repo, "github.com/", "github.com:"))"
+
+    # The upstream URL to which we push new content and the ssh decryption commands.
```
Comment (mortenpi) [inline src/Documenter.jl]: This comment is on the wrong line. Should go before the `upstream` assignment.

**Comment by mortenpi** [comment]
Thanks @rofinn!

**Pull request merged**